File: MultiLanguageSandbox/src/test/deps/pascal/29.pas
{$ASSERTIONS ON}
(*
  Evaluate the grade for an input integer.
  Given an integer score, this function returns a character representing the grade.
  An 'A' grade is assigned for scores between 90 and 100 (inclusive),
  while a 'B' grade is given for all other scores.
  
  Example usage:
    >>> evaluate_integer_grade(90)
    A
*)

function evaluate_integer_grade(score: integer): char;
begin
  if (score >= 90) and (score <= 100) then
    evaluate_integer_grade := 'A'
  else
    evaluate_integer_grade := 'B';
end;
procedure testEvaluateIntegerGrade;
begin
  Assert(evaluate_integer_grade(90) = 'A');
  Assert(evaluate_integer_grade(89) = 'B');
  Assert(evaluate_integer_grade(95) = 'A');
  Assert(evaluate_integer_grade(100) = 'A');
  Assert(evaluate_integer_grade(101) = 'B');
end;

begin
  testEvaluateIntegerGrade;
end.

File: MultiLanguageSandbox/src/test/deps/typescript/29.ts

/**
 * Evaluate the grade for an input integer.
 * @param score - The integer score to be evaluated.
 * @returns The grade corresponding to the input score.
 *          If the score is between 90 and 100 (inclusive), returns 'A'.
 *          Otherwise, returns 'B'.
 * 
 * Examples:
 * evaluateIntegerGrade(90) // returns 'A'
 *
 */

function evaluateIntegerGrade(score: number): string {
    if (score >= 90 && score <= 100) {
        return 'A';
    } else {
        return 'B';
    }
}
const testEvaluateIntegerGrade = (): void => {
    console.assert(evaluateIntegerGrade(90) === 'A', "Expected 'A' for score 90");
    console.assert(evaluateIntegerGrade(89) === 'B', "Expected 'B' for score 89");
    console.assert(evaluateIntegerGrade(95) === 'A', "Expected 'A' for score 95");
    console.assert(evaluateIntegerGrade(100) === 'A', "Expected 'A' for score 100");
    console.assert(evaluateIntegerGrade(101) === 'B', "Expected 'B' for score 101");
    
    // console.log("All tests passed");
};

testEvaluateIntegerGrade();